src/worker.js

```diff
@@ -21,9 +21,48 @@ export default {
     if (provider == "token") return await tokenFromCredential(url.searchParams.get("credential"), env.AIPIPE_SECRET);
 
     // Check if the URL matches a valid provider. Else let the user know
-    if (!providers[provider] && provider != "usage" && provider != "admin")
+    if (!providers[provider] && provider != "usage" && provider != "admin" && provider != "proxy")
       return jsonResponse({ code: 404, message: `Unknown provider: ${provider}` });
 
+    // Handle proxy requests
+    if (provider === "proxy") {
+      const targetUrl = request.url.split("/proxy/")[1]; // Remove /proxy/ from the path
+      if (!targetUrl.startsWith("http")) {
+        return jsonResponse({ code: 400, message: "URL must begin with http" });
+      }
+
+      const controller = new AbortController();
+      const timeoutId = setTimeout(() => controller.abort(), 30000); // 30 seconds timeout
+
+      try {
+        const response = await fetch(targetUrl, {
+          method: request.method,
+          headers: request.headers,
```

Comment:
Skip these headers when making the request
  - host
  - origin
  - connection
  - keep-alive

Strip these headers when sending the response
  - connection
  - content-encoding
  - content-length
  - host
  - transfer-encoding
  # OpenRouter sends security headers that we normally set. Skip them.
  - content-security-policy
  - access-control-allow-headers
  - access-control-allow-methods
  - access-control-allow-origin
  # Skip all headers that we override
  - access-control-expose-headers

In addCors(), add access-control-expose-headers: *